CoffeeMachineSimulator/CoffeeMachineSimulator.Tests/Services/CoffeeServiceTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿using CoffeeMachineSimulator.Services.Models;
+using CoffeeMachineSimulator.Services.Services;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CoffeeMachineSimulator.Tests.Services
+{
+    [TestFixture]
+    public class CoffeeServiceTests
+    {
+        private CoffeeService coffeeService;
+
+        [SetUp]
+        public void SetUp()
+        {
+            coffeeService = new CoffeeService();
+        }
+
+        [Test]
+        public void GetCoffees_Returns_ListOfCoffeeModels()
+        {
+            var coffeesReturned = coffeeService.GetCoffees();
+
+            Assert.IsNotNull(coffeesReturned);
+            Assert.IsTrue(coffeesReturned.Any());
+            Assert.IsInstanceOf(typeof(List<CoffeeModel>), coffeesReturned);
+            Assert.IsTrue(coffeesReturned.Any(coffeeModel => coffeeModel.Id != Guid.Empty && !string.IsNullOrEmpty(coffeeModel.Name)));
+        }
+
+        [Test]
+        public void AddNullCoffee_Throws_NewException()
+        {
+            var ex = Assert.Throws<Exception>(() => coffeeService.AddCoffee(null));
+            
+            Assert.AreEqual(ex.Message, "You should not add null entries!");
+        }
+
+        [Test]
+
+        //TODO: Add Unit test for Remaining add cases
+        public void AddExistingCoffee_Throws_NewException()
+        {            
+            var coffeeadded = new CoffeeModel(); 
+            coffeeadded = coffeeService.GetCoffees()[0];
+            var ex= Assert.Throws<Exception>(() => coffeeService.AddCoffee(coffeeadded));
+
+            Assert.AreEqual(ex.Message, "Already exists an entry with the same Id");
+        }
+
+        [Test]
+        public void AddEmptyFieldCoffee_Throws_NewException()
+        {
+            var marime = coffeeService.GetCoffees().Count;
+            var coffeeadded = new CoffeeModel(); 
+            coffeeadded.Id = Guid.NewGuid(); 
+            coffeeadded.Name = "Nume";
+            coffeeService.AddCoffee(coffeeadded);
+
+            Assert.AreEqual(marime, coffeeService.GetCoffees().Count);
```

Comment:
You could make separate tests for each of this check (AddCoffeeWithExistingName_DoesNotAddNewCoffee)